src/transforms/aws_ec2_metadata.rs

```diff
@@ -132,7 +131,7 @@ impl TransformConfig for Ec2Metadata {
         let host = self
             .host
             .clone()
-            .map(|s| Uri::from_shared(s.into()).unwrap())
+            .map(|s| Uri::from_maybe_shared(s).unwrap())
```

Comment:
Can be done without clone (but `HOST.clone()` better move to closure):
```rust
        let host = self
            .host
            .as_ref()
            .map(|s| s.parse::<Uri>().unwrap())
            .unwrap_or_else(|| HOST.clone());
```
Another thing, that it would be better handle error instead unwrap (not sure that creating snafu error from sinks mod is correct):
```rust
            .map(|s| {
                s.parse::<Uri>()
                    .map_err(|source| crate::sinks::BuildError::UriParseError2 { source })
            })
            .unwrap_or_else(|| Ok(HOST.clone()))?;
```

Reply:
I think using a sink error in a transform is suspect at best, although this is exactly the kind of error we want to use. It might make sense to have a crate-level set of errors for some of these general errors, but that's outside the scope of this PR. Please open an issue to discuss it.